go/teams/loader.go

```diff
@@ -108,6 +114,74 @@ func (l *TeamLoader) HintLatestSeqno(ctx context.Context, teamID keybase1.TeamID
 	return nil
 }
 
+type nameLookupBurstCacheKey struct {
+	teamName keybase1.TeamName
+	public   bool
+}
+
+func (n nameLookupBurstCacheKey) String() string {
+	return fmt.Sprintf("%s:%v", n.teamName.String(), n.public)
+}
+
+// Resolve a team name to a team ID.
+// Will always hit the server for subteams. The server can lie in this return value.
+func (l *TeamLoader) ResolveNameToIDUntrusted(ctx context.Context, teamName keybase1.TeamName, public bool, allowCache bool) (id keybase1.TeamID, err error) {
+
+	defer l.G().CVTrace(ctx, libkb.VLog0, fmt.Sprintf("resolveNameToUIDUntrusted(%s,%v,%v)", teamName.String(), public, allowCache), func() error { return err })()
+
+	// For root team names, just hash.
+	if teamName.IsRootTeam() {
+		return teamName.ToTeamID(public), nil
+	}
+
+	if !allowCache {
+		return resolveNameToIDUntrustedAPICall(ctx, l.G(), teamName, public)
+	}
+
+	var idVoidPointer interface{}
+	key := nameLookupBurstCacheKey{teamName, public}
+	idVoidPointer, err = l.nameLookupBurstCache.Load(ctx, key, l.makeNameLookupBurstCacheLoader(ctx, l.G(), key))
+	if err != nil {
+		return keybase1.TeamID(""), err
+	}
+	if idPointer, ok := idVoidPointer.(*keybase1.TeamID); ok {
+		id = *idPointer
+	} else {
+		return keybase1.TeamID(""), errors.New("bad cast out of nameLookupBurstCache")
+	}
+	return id, nil
+}
+
+func resolveNameToIDUntrustedAPICall(ctx context.Context, g *libkb.GlobalContext, teamName keybase1.TeamName, public bool) (id keybase1.TeamID, err error) {
```

Comment:
this and `LoaderContextG.resolveNameToIDUntrusted` are now duplicate

Reply:
oh i didn't remove it?